server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -1104,16 +1104,71 @@ public String getOffsetFieldName() {
         return offsetsFieldName;
     }
 
+    /**
+     * Class that holds either a UTF-16 String or a UTF-8 BytesRef, and lazily converts between the two.
+     */
+    private static class RawString {
```

Comment:
Can this be a record?

Reply:
Possibly, but it would be a bit tough since records are immutable. The lazy conversion works by only setting the `bytesValue` or `stringValue` once they've been converted, which wouldn't work if the properties were `final`.

Reply:
Ok, I thought falsely assumed that making `RawString`'s properties final would be straightforward. In that case turning this class into a record doesn't make sense.